File: oop/cont_bancar.py
class ContBancar:
    # constructor
    def __init__(self, titularCont, iban):
        # atribute, fielsd
        self.titularCont = titularCont
        self.iban = iban
        self.sold = 0
        self.activ = False
        self.pin = 7777
        self.incercari_activare = 0

    # metode
        # nu se poate defini o metoda in alta metoda
        # se poate apela o metoda in alta metoda

    def salut(self):
        print(f'Buna {self.titularCont}')

    def interogareSold(self):
        print(f'Titular: {self.titularCont}')
        print(f'IBAN: {self.iban}')
        print(f'Sold: {self.sold}')
        print(f'Activ: {self.activ}')
        print(f'Nr de incercari gesite: {self.incercari_activare}')
        print('--------------------------------')

    def activareCont(self, pin_utilizator):
        # modifica activ in true doar daca pinul este corect
        self.salut()
        if pin_utilizator == self.pin:
            print(f'Activat cu succes.')
            self.activ = True
        else:
            print(f'PIN gresit')
            self.incercari_activare += 1 # incrementeaza (este operator de atribuire)
                # se poate scrie si self.incercari_activare = self.incercari_activ1
        print('--------------------------------')

    def alimentareCont(self, suma):
        self.sold += suma # se ia suma si se adauga la sold
        self.salut()
        print(f'Ai depus cu succes suma de {suma}. Ai in cont suma de {self.sold}')
        print('--------------------------------')

    def plataCard(self, suma): # numconteaza faptul ca au aceeasi denumire - fiecare activeaza doar in metoda ei si nu stiu una de cealalta
        self.salut()
        # pot sa cheltuiesc doar ce am
        # daca suma cheltuita <= sold
        if suma <= self.sold:
            self.sold -= suma # se ia soldul si se scade suma
                # se poate scrie si self.sold = self.sold - suma
            print(f'Ai cheltuit {suma}')
            print(f'Mai ai in cont {self.sold}')
        else:
            print(f'Fonduri insuficiente')
        print('--------------------------------')

File: src/oop/ContBancar.java
package oop;

public class ContBancar {
    // program files - definim logica unui cont bancar

    // OOP = object oriented programing
    // POO = programare orientata pe obiecte

    // o clasa = un tip nou
    // este definitia unui concept (o reteta)
    // Ex: reteta pt paste carbonara, ADN-ul uman, masina

    // un obiect = instanta unei clare - de exemplu pastele carbonara
    // Ex: 10 obiecte de tip paste carbonara, omul, masina in sine

    // fields = prorpietati = atribute
    // variabile
    // ex: culoare, marca, model, consum, pret, esteOprita

    //metode = actiuni ce pot fi facute de obiecte
    // functii
    // ex: accelereaza(), franeaza(), deschideUsa(), vopsire()

    // proprietatile, fields
    String titularCont;
    String iban;
    double sold = 0;
    boolean activ = false;
    private int pin = 7777;
    int incercari_activare = 0;

    // constructor = are rolul de a impune date de start
    // ca si * din formulare, required fields
    public ContBancar(String titularCont, String iban) {
        this.titularCont = titularCont;
        this.iban = iban;
    }

    // metode
    public void interogareSold() {
        System.out.println("Titular: " + this.titularCont);
        System.out.println("IBAN: " + this.iban);
        System.out.println("Sold: " + this.sold);
        System.out.println("Activ: " + this.activ);
        System.out.println("Nr de incercari gesite: " + this.incercari_activare);
        System.out.println("--------------------------------");
    }

    public void activareCont(int pinUtilizator) {
        //modifica activ in true doar daca pinul este corect
        System.out.println("Buna " + this.titularCont + ".");
        if (pinUtilizator == this.pin) {
            System.out.println("Activat cu succes.");
            this.activ = true;
        } else {
            System.out.println("PIN gresit");
            this.incercari_activare++; //incrementeaza
        }
        System.out.println("--------------------------------");
    }

    public void alimentareCont(double suma_depusa) {
        //la ce aveam in cont (this.sold) se mai adauga suma_depusa
        this.sold = this.sold + suma_depusa;
        System.out.println("Buna " + this.titularCont + ".");
        System.out.println("Ati depus cu succes suma de " + suma_depusa + ". Aveti in cont suma de " + this.sold + ".");
        System.out.println("--------------------------------");
    }

    public void plataCard(double suma_cheltuita) {
        System.out.println("Buna " + this.titularCont + ".");
        // pot sa cheltuiesc doar ce am
        // daca suma ch <= sold
        if (suma_cheltuita <= this.sold) {
            //dispar banii din cont
            this.sold = this.sold - suma_cheltuita;
            System.out.println("Ai cheltuit " + suma_cheltuita);
            System.out.println("Mai ai " + this.sold);
        } else {
            System.out.println("Fonduri insuficiente");
        }
        System.out.println("--------------------------------");
    }
}
